tests/cpp-tests/Classes/BugsTest/Bug-15776.cpp

```diff
@@ -0,0 +1,26 @@
+//
+//  Bug-15776.cpp
+//  cocos2d_tests
+//
+//  Created by Ricardo Quesada on 6/17/16.
+//
+//
+
+#include "Bug-15776.h"
+
+
+USING_NS_CC;
+
+bool Bug15776Layer::init()
+{
+    if (BugsTestBase::init())
+    {
+        cocos2d::Image *cocos2dxImage = new cocos2d::Image();
+        cocos2dxImage->initWithImageData(nullptr, 0);
+        // should not crash. invalid cocos2dImage
+        auto texture2d = Director::getInstance()->getTextureCache()->addImage(cocos2dxImage, "unused");
+        return texture2d;
```

Comment:
@minggo correct. I haven't "fixed" the bug. What I added was an assert so that it crashes when it should crash and not somewhere else.

But I think in v4 we should change the policy of cocos2d-x, and it should never crash, even when it receives invalid data. When it receives invalid data it should use a default one.